.github/workflows/wheel.yml

```diff
@@ -0,0 +1,414 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright Contributors to the OpenImageIO Project.
+#
+
+name: Wheel
+
+permissions:
+  contents: read
+  id-token: write
+
+on:
+  push:
+    # Workflow run on tags for v2.6 and v3 only.
+    tags:
+      - v2.6.*
+      - v3.*
+  pull_request:
+    # Workflow run on pull_request only when related files change.
+    branches-ignore:
+      - dev-*
+      - lg-*
+      - rtd
+      - sonar-1
+    paths:
+      - .github/workflows/wheel.yml
+      - pyproject.toml
+      - src/python/OpenImageIO/*.py
+      - src/cmake/pythonutils.cmake
+  schedule:
+    # Nightly build
+    - cron:  "0 8 * * *"
+  workflow_dispatch:
+    # This allows manual triggering of the workflow from the web
+
+jobs:
+  # Linux jobs run in Docker containers (manylinux), so the latest OS version
+  # is OK. macOS and Windows jobs need to be locked to specific virtual
+  # environment versions to mitigate issues from OS updates, and will require
+  # maintenance as OS versions are retired.
+
+  # ---------------------------------------------------------------------------
+  # Source Distribution
+  # ---------------------------------------------------------------------------
+
+  sdist:
+    name: Build SDist
+    runs-on: ubuntu-latest
+    # Don't run on OIIO forks
+    if: |
+      github.event_name != 'schedule' ||
+      github.repository == 'AcademySoftwareFoundation/OpenImageIO'
```

Comment:
You sure this logic is correct? You definitely only want it to run on the main repo, not people's forks. But doesn't this say that as long as it's not the scheduled event, it will run on forks?

Reply:
Hmmm.... that's a good point. I think you're right... if anything, we'd want an `&&` operand (which I guess would negate the cron stuff); or we'd want to use only the github.repository bit. 

I based this entire workflow off of OCIO's [wheel_workflow.yml](https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/main/.github/workflows/wheel_workflow.yml), cuz I barely know what I'm doing with github actions -- maybe @remia can elaborate?

Reply:
I think our goal was to allow the workflow to run on people's forks if trigger by a push / PR but not by the nightly cron job, if that makes sense.

Reply:
I think you definitely want the second clause (repository) to keep it from running on everybody's fork push or whatever. But I don't think you want the first clause at all, because we want it to run for all of scheduled, PRs, and pushes to the official repo, right?

Reply:
It can be useful to allow the workflow to run on people forks before submitting on the main repo, especially for first time contributors that have to wait for approval every time they push. This is only for building the wheel obviously.

Reply:
I think @zachlewis set it similar to OCIO where only PRs that are updating Python related code will trigger the wheel build. Combined with the nightly (which run only on the main repo) this has been sufficient for OCIO needs at least.

Reply:
On it!